kernel/fmt.cc

```diff
@@ -740,7 +740,7 @@ std::string Fmt::render() const
 				log_assert(part.width == 0 || part.padding != '\0');
 				if (part.justify == FmtPart::RIGHT && buf.size() < part.width) {
 					size_t pad_width = part.width - buf.size();
-					if (part.padding == '0' && (buf.front() == '+' || buf.front() == '-')) {
+					if (part.padding == '0' && (!buf.empty() && (buf.front() == '+' || buf.front() == '-'))) {
```

Comment:
As a nit there seems to be one set of parentheses too many, otherwise LGTM, great to have this fixed